crates/arroyo-connectors/src/iggy/mod.rs

```diff
@@ -0,0 +1,624 @@
+            let password = options.pull_str("password")?;
+            IggyConfigAuthentication::UsernamePassword {
+                username,
+                password: VarStr::new(password),
+            }
+        } else {
+            IggyConfigAuthentication::None {}
+        };
+
+        let transport_protocol = match transport.as_str() {
+            "http" => TransportProtocol::Http,
+            _ => bail!("Invalid transport protocol"),
+        };
+
+        let config = IggyConfig {
+            endpoint: endpoint.clone(),
+            transport: transport_protocol,
+            authentication: authentication.clone(),
+        };
+
+        let stream = options.pull_str("stream")?;
+        let topic = options.pull_str("topic")?;
+
+        let table_type = if options.pull_bool("is_source")? {
+            let offset = match options.pull_str("offset")?.as_str() {
+                "earliest" => "earliest",
+                "latest" => "latest",
+                _ => bail!("Invalid offset value"),
+            };
+
+            let consumer_id = options.pull_u64("consumer_id")? as u32;
+            let partition_id = options.pull_u64("partition_id")? as u32;
+            let auto_commit = options.pull_bool("auto_commit")?;
+
+            TableType::Source {
+                offset: match offset {
+                    "earliest" => SourceOffset::Earliest,
+                    "latest" => SourceOffset::Latest,
+                    _ => bail!("Invalid offset value"),
+                },
+                consumer_id: consumer_id as i64,
+                partition_id: partition_id as i64,
+                auto_commit,
+            }
+        } else {
+            let partitioning = match options.pull_str("partitioning")?.as_str() {
+                "partition_id" => "partition_id",
+                "balanced" => "balanced",
+                _ => bail!("Invalid partitioning value"),
+            };
+
+            let partition_id = options.pull_opt_u64("partition_id")?.map(|id| id as u32);
+
+            TableType::Sink {
+                partitioning: match partitioning {
+                    "partition_id" => SinkPartitioning::PartitionId,
+                    "balanced" => SinkPartitioning::Balanced,
+                    _ => bail!("Invalid partitioning value"),
+                },
+                partition_id: partition_id.map(|id| id as i64),
+            }
+        };
+
+        let table = IggyTable {
+            stream,
+            topic,
+            type_: table_type,
+        };
+
+        self.from_config(None, name, config, table, schema)
+    }
+
+    fn from_config(
+        &self,
+        id: Option<i64>,
+        name: &str,
+        config: Self::ProfileT,
+        table: Self::TableT,
+        schema: Option<&ConnectionSchema>,
+    ) -> Result<ArroyoConnection> {
+        let schema = schema.cloned().unwrap_or_else(|| ConnectionSchema {
```

Comment:
This needs to use the actual passed in schema. See examples from the kafka connector.